src/ZkFold/Symbolic/Cardano/Contracts/RandomOracle.hs

```diff
@@ -25,23 +25,11 @@ type Tx context = Transaction 1 0 2 Tokens 1 () context
 hash :: forall context x . (Symbolic context, MiMCHash (BaseField context) context x )=> x -> FieldElement context
 hash = mimcHash @(BaseField context) mimcConstants zero
 
-type Sig context =
+randomOracle :: forall context .
     ( Symbolic context
     , FromConstant (BaseField context) (FieldElement context)
-    , MultiplicativeMonoid (UInt 64 Auto context)
-    , Eq (Bool context) (FieldElement context)
-    , Eq (Bool context) (UInt 64 Auto context)
-    , Eq (Bool context) (ByteString 224 context)
-    , Eq (Bool context) (ByteString 256 context)
-    , Extend (ByteString 224 context) (ByteString 256 context)
-    , Extend (AssetName context) (AssetName context)
-    , BinaryExpansion (FieldElement context)
     , Bits (FieldElement context) ~ FieldElementBits context
```

Comment:
Maybe it is better to remove `FieldElementBits` alias and this constraint and adapt the contract code for variable number of bits in the basefield? @vlasin, what do you think?

Reply:
@vlasin, maybe we can do it in a separate PR then? This is not in scope of this PR IMO